combit.ListLabel.SchemaAwareJsonDataProvider/SchemaAwareJsonDataProvider.cs

```diff
@@ -35,6 +38,12 @@ public string Schema
                 _schemaLocation = value;
                 var fileProvider = SchemaFileProvider ?? Options?.FileProvider ?? new NetworkFileProvider();
                 _schema = JsonSchema.FromJsonAsync(fileProvider.ReadAsString(_schemaLocation)).Result;
+
+                RootTableName = _schema.Title ?? RootTableName;
+                if (Data.IsArray)
+                {
+                    ArrayValueName = _schema.Definitions.FirstOrDefault().Key ?? ArrayValueName;
```

Comment:
Not sure as schematized JSON is not my primary science 😁 - could FirstOrDefault() also return `null` here? In that case, it might be better to use `_schema.Definitions.FirstOrDefault()?.Key ?? ArrayValueName`?

Reply:
Resolved by your commit [588ee9a](https://github.com/combit/NETDataProviders/commit/588ee9a41eff5706bde68487874f4600eef0f04f)

Reply:
`IDictionary<string, JsonSchema>.FirstOrDefault()` will return a default `KeyValuePair<string, JsonSchema>` with `string` defaulting to `null`. However, `Definitions` can be null, so I added a null check there.